thunder/torch/__init__.py

```diff
@@ -5331,6 +5331,215 @@ def wait(slf) -> None:
+
+    return wrapper, differentiable_args
+
+
+def register_default_torch_ops():
+    from thunder.torch import default_torch_ops
+
+    for m, fns in default_torch_ops.torch_auto_registered_ops.items():
+        for fn in fns:
+            register_default_torch_op(fn, meta_adaptor(fn), m)
+
+
+def register_default_torch_op(torchfn: Callable, fn_meta: Callable, m):
+    _fn = langctx(Languages.TORCH)(fn_meta)
+    sym = Symbol(
+        name=torchfn.__name__,
+        meta=_fn,
+        id=f"{m.__name__}.{torchfn.__name__}",
+    )
+    _torch_to_thunder_function_map[torchfn] = sym
+    from thunder.executors.torchex import _always_executable, ex
+
+    op = ex.register_operator(torchfn.__name__, module=m, meta=fn_meta)
+    ex.register_implementation(sym, op, checker=_always_executable)
+
+    from thunder.core.transforms import augmented_forward_impls, backward_impls
+
+    augmented_forward_impls[sym.id] = augmented_forward_adaptor(op)
+
+    def _vjp_impl(residules, *gs) -> torch.Tensor:
+        inp_args, inp_kwargs = residules
+
+        wrapped_func, diff_args = _make_differentiable_wrapper(torchfn, *inp_args, **inp_kwargs)
+        _, outs = torch.autograd.functional.vjp(wrapped_func, diff_args, v=gs)
+
+        from thunder.core.pytree import tree_unflatten, tree_flatten
+
+        flat_inp_args, inp_spec = tree_flatten(inp_args)
+        iter_out = iter(outs if isinstance(outs, Sequence) else (outs,))
+        outs = tuple(next(iter_out) if _is_differentiable(arg) else None for arg in flat_inp_args)
+        return tree_unflatten(outs, inp_spec)
+
+    bwd_op = ex.register_operator(torchfn.__name__ + "_vjp", meta=backward_adaptor(torchfn), fn=_vjp_impl)
+    ex.register_implementation(bwd_op.id, bwd_op, checker=_always_executable)
+    backward_impls[sym.id] = bwd_op
+
+
+def _get_fake_arg(inp: Any, fake_mode):
+    from thunder.core.devices import to_torch_device
+    from thunder.core.proxies import TupleProxy, ListProxy, DictProxy
+
+    if inp is None:
+        return inp
+    if isinstance(inp, NumberProxy):
+        if inp.value == None:
+            raise NotImplementedError("Unsupported for NumberProxy.value=None")
+        else:
+            return inp.value
+    elif isinstance(inp, TensorProxy):
+        return fake_mode.from_tensor(
+            torch.empty(
+                inp.shape,
+                dtype=_thunder_to_torch_dtype_map[inp.dtype],
+                requires_grad=inp.requires_grad,
+                device=to_torch_device(inp.device),
+            )
+        )
+    elif isinstance(inp, (TupleProxy, ListProxy, DictProxy)):
+        return inp._value
+    elif isinstance(inp, (torch.dtype, torch.device, torch.Size, torch.memory_format, str, bool, int, float, complex)):
+        return inp
+    elif isinstance(inp, devices.Device):
+        return to_torch_device(inp)
+    elif isinstance(inp, dtypes.dtype):
+        return to_torch_dtype(inp)
+    else:
+        from builtins import type
```

Comment:
Q: would this import be strongly required?

Reply:
ah I see. I have no nice alternatives out of my mind

Reply:
Something for us to think about in the future:

In general, we don't like using names that conflict with Python builtins, but this isn't always possible. If there are multiple `type` functions, then we should probably note the confusion and try to explicitly disambiguate them by calling the fully qualified name of each function, like `builtins.type`. We don't need to do anything about this today, it's just an idea for how we write maintainable and readable code in the future.